src/main/java/org/jenkinsci/plugins/gitclient/Git.java

```diff
@@ -136,6 +141,29 @@ public GitClient invoke(File f, VirtualChannel channel) throws IOException, Inte
     }
 
     /**
+     * Retrieve the {@link Launcher} to be used to run git cli.
+     */
+    private @CheckForNull Launcher getLauncher() {
```

Comment:
If `repository` is a `FilePath`, then yes that would be the right approach to get a default `Launcher` (missing any decorations etc. that might have been passed in).